helm/cluster/templates/_helpers.tpl

```diff
@@ -151,4 +151,115 @@ Where `data` is the data to hash and `global` is the top level scope.
+  {{- $providerIntegration = $.Values.providerIntegration }}
+{{- /* Case 2: template is called from a parent chart that passes Helm values to cluster chart in .Values.cluster */}}
+{{- else if ($.Values.cluster).providerIntegration }}
+  {{- $providerIntegration = $.Values.cluster.providerIntegration }}
+{{- end }}
+{{- /* Create $.GiantSwarm object where we put custom Giant Swarm vars */}}
+{{- if not $.GiantSwarm }}
+  {{- $_ := set $ "GiantSwarm" dict }}
+{{- end }}
+{{- /* Finally set "UseReleases" property that can be used in other templates */}}
+{{- $_ := set $.GiantSwarm "providerIntegration" $providerIntegration }}
+{{- end }}
+
+{{- define "cluster.internal.get-internal-values" }}
+{{- $internalValues := dict }}
+{{- /* Case 1: template is called from the cluster chart itself */}}
+{{- if and (eq $.Chart.Name "cluster") $.Values.internal }}
+  {{- $internalValues = $.Values.internal }}
+{{- /* Case 2: template is called from a parent chart that passes Helm values to cluster chart in .Values.cluster */}}
+{{- else if ($.Values.cluster).internal }}
+  {{- $internalValues = $.Values.cluster.internal }}
+{{- end }}
+{{- /* Create $.GiantSwarm.internal object */}}
+{{- if not $.GiantSwarm }}
+  {{- $_ := set $ "GiantSwarm" dict }}
+{{- end }}
+{{- /* Finally set "UseReleases" property that can be used in other templates */}}
+{{- $_ := set $.GiantSwarm "internal" $internalValues }}
+{{- end }}
+
+{{- define "cluster.internal.get-release-resource" }}
+{{- $_ := include "cluster.internal.get-internal-values" $ }}
+{{- $renderWithoutReleaseResource := ((($.GiantSwarm.internal).ephemeralConfiguration).offlineTesting).renderWithoutReleaseResource | default false }}
+{{- $releaseVersion := $.Values.global.release.version | trimPrefix "v" }}
+{{- $releaseVersion = printf "v%s" $releaseVersion }}
+{{- $release := lookup "release.giantswarm.io/v1alpha1" "Release" "" $releaseVersion }}
+{{- if $release }}
+  {{- $_ := set $.GiantSwarm "Release" $release }}
+{{ else if not $renderWithoutReleaseResource }}
+  {{- fail (printf "Release resource '%s' not found" $releaseVersion) }}
+{{- end }}
+{{- end }}
+
+{{- define "cluster.app.version" }}
+{{- $appVersion := "N/A" }}
+{{- $_ := (include "cluster.internal.get-release-resource" $) }}
+{{- if $.GiantSwarm.Release }}
+{{- range $_, $app := $.GiantSwarm.Release.spec.apps }}
+{{- if eq $app.name $.appName }}
+{{- $appVersion = $app.version }}
+{{- end }}
+{{- end }}
+{{- end }}
+{{- $appVersion }}
+{{- end }}
+
+{{- define "cluster.component.version" }}
+{{- $componentVersion := "N/A" }}
+{{- $_ := (include "cluster.internal.get-release-resource" $) }}
+{{- if $.GiantSwarm.Release }}
+{{- range $_, $component := $.GiantSwarm.Release.spec.components }}
+{{- if eq $component.name $.componentName }}
+{{- $componentVersion = $component.version }}
+{{- end }}
+{{- end }}
+{{- end }}
+{{- $componentVersion }}
+{{- end }}
+
+{{- define "cluster.component.kubernetes.version" }}
```

Comment:
Helper to get Kubernetes version:
- For providers that use Release CRs, it gets k8s version from the release CR.
- For providers that do not use Release CRs, it gets k8s version from the Helm value like before.

This way we can release cluster chart and during the transition phase we can use the same cluster chart version across different providers, where some providers use new releases and others do not.